fw/http2.c

```diff
@@ -283,16 +283,21 @@ tfw_h2_remove_idle_streams(TfwH2Ctx *ctx, unsigned int id)
 	 */
 	assert_spin_locked(&((TfwConn *)conn)->sk->sk_lock.slock);
 
-	list_for_each_entry_safe_reverse(stream, tmp, &ctx->idle_streams.list,
-					 hcl_node)
-	{
-		if (id <= stream->id)
-			break;
+	/*
+	 * I add this because we get BUG: kernel NULL pointer dereference,
+	 * in case ERROR: cannot establish a new h2 connection
+	 */
+	if (ctx->idle_streams.num)
```

Comment:
This is unnecessary, as I think it should not be needed on the current master. If not please implement tests when it is necessary

Reply:
Yes. I agree, I wrote it before changing on the master.